src/VisualStudio/VisualStudioDiagnosticsToolWindow/Loggers/OutputWindowLogger.cs

```diff
@@ -19,19 +19,21 @@ internal sealed class OutputWindowLogger : ILogger
     {
         private readonly Func<FunctionId, bool> _loggingChecker;
 
-        public OutputWindowLogger()
-            : this((Func<FunctionId, bool>)null)
-        {
-        }
+        private readonly IAsynchronousOperationListener _asyncListener;
+        private readonly ServiceBrokerClient _serviceBrokerClient;
+        private readonly IThreadingContext _threadingContext;
 
-        public OutputWindowLogger(IGlobalOptionService optionService)
-            : this(Logger.GetLoggingChecker(optionService))
-        {
-        }
-
-        public OutputWindowLogger(Func<FunctionId, bool> loggingChecker)
+        public OutputWindowLogger(Func<FunctionId, bool> loggingChecker, IAsynchronousOperationListenerProvider asyncListenerProvider,
+            IThreadingContext threadingContext, IBrokeredServiceContainer brokeredServiceContainer)
         {
             _loggingChecker = loggingChecker;
+
+            Assumes.Present(brokeredServiceContainer);
+            var serviceBroker = brokeredServiceContainer.GetFullAccessServiceBroker();
+
+            _asyncListener = asyncListenerProvider.GetListener(FeatureAttribute.OutputWindowLogger);
+            _threadingContext = threadingContext;
+            _serviceBrokerClient = new ServiceBrokerClient(serviceBroker, _threadingContext.JoinableTaskFactory);
```

Comment:
Is there any overhead of creating multiple Clients, or is there some benefit of sharing things?

Reply:
Good question, I'll defer to Andrew

@AArnott is there any issue with sharing a single service broker client to retrieve various distinct services?  Is there any benefit to doing that?

Reply:
A `ServiceBrokerClient` is totally fine with you requesting several services from it. It's optimized for that. What you *shouldn't* do is share your `ServiceBrokerClient` with a lot of users/classes if you can avoid it. The `AvailabilityChanged` event can get noisy if it's raising events to a lot of listeners for services that aren't related to an individual handler. So basically if you're setting up multiple handlers for that event to inform different classes that all consume from the same `ServiceBrokerClient`, consider creating a separate `ServiceBrokerClient` (and each with its own `IServiceBroker` passed to it) for each of these classes.